doc/news/changes/incompatibilities/20210623Munch

```diff
@@ -0,0 +1,5 @@
+Deprecated: The functions MGTwoLevelTransfer::reinit_geometric_transfer()
+and MGTwoLevelTransfer::reinit_polynomial_transfer(). Please, use instead the
+unified function MGTwoLevelTransfer::reinit().
+<br>
+(Peter Munch, 2021/06/23)
```

Comment:
I do not like that we deprecate an interface introduced in a release a month ago, and in particular not during a cycle where we still have several aspects of multigrid under heavy development (so we might change again before the next release). I like the self-documenting aspect of the old interfaces, as they show the user at first sight what kind of operations are supported. Of course, I would be happy to be wrong in that we never need these names for feature documentation. From the side of the implementation, we could of course do as this PR suggests and let everything go through the same interface, as it would help to make the move in the future.  Either way, I see no pressing to deprecate these functions now.